pkg/daemon/ceph/rgw/objectstore.go

```diff
@@ -219,6 +219,21 @@ func deletePools(context *Context, lastStore bool) error {
 		}
 	}
 
+	// Delete erasure code profile if any
+	erasureCodes, err := ceph.ListErasureCodeProfiles(context.context, context.ClusterName)
+	if err != nil {
+		return fmt.Errorf("failed to list erasure code profiles for cluster %s: %+v", context.ClusterName, err)
+	}
+	objectStoreErasureCode := fmt.Sprintf("%s_ecprofile", context.Name)
```

Comment:
Note that this implicitly is only going to work for the data pools EC profile.  That's OK because according to https://rook.io/docs/rook/master/object-store-crd.html#pools, the metadata pools only support replication, but it may be good to add a comment somewhere here noting that.

Reply:
True enough but we're going to fix this in #1328 I hope.

Reply:
i couldnt figure out a way to determine whether the pool is of type ErasureCoded or not. Since i dont have the pool spec reference. So i thought this was the next best thing to do. Let me know if you know of a better. I am very new to this code.

Reply:
yes, i think the approach here is OK (since you don't have a pool spec at this point), but it would be nice to just add a simple comment mentioning that this only cleans up the EC profile for the data pool because metadata pools don't support EC.